src/android/ContactAccessorSdk5.java

```diff
@@ -1027,7 +1027,7 @@ else if (accounts.length > 1) {
     private String modifyContact(String id, JSONObject contact, String accountType, String accountName) {
         // Get the RAW_CONTACT_ID which is needed to insert new values in an already existing contact.
         // But not needed to update existing values.
-        int rawId = (Integer.valueOf(getJsonString(contact, "rawId"))).intValue();
+        String rawId = getJsonString(contact, "rawId");
```

Comment:
From my experiments this works both with `int` and `String`. My guess is that ContentProvider casts types implicitly, when needed.
Also, according to the [same documentation](http://developer.android.com/intl/ru/reference/android/provider/ContactsContract.Data.html) `id` is also of type `long`, but is has been used here as `String` always. So IMO it doesn't matter.